### PR TITLE
working CAS for professors and students

### PR DESCRIPTION
**Pull request by longn5**
CAS now checks if a professor is logging in.  If it is a professor then the professor dashboard is shown. 
 If the user is not a professor, it checks if they are in the student table.  If they are in the student table they are routed to the review.  If not then they are routed back to the homepage.

To test this, you need to add your CAT username into both the professor and student table.  Check to see if the professor dashboard shows up when logging in.  After this check, remove your username from the professor table.  Check login again, this time you will be routed to the student review form.

### CODE REVIEW

**Review by pdx-cs-lbryan — CHANGES_REQUESTED** [review]
* your CAT login name is currently in the DB, you may want to remove it

* submitting reviews currently doesn't work:
When logging in as a student, log in will take the user to /dashboard/, where they'll see the review
However, when they submit the review, because they are on /dashboard/, it is using prof_dashboard.py's post method. Thus, on submission, there's an exception that there is no 'session_id' in the form (line 61).

* form.py doesn't seem to have all instances of getting the student id changed to account for log in. A quick fix for now would be to change it in the 'get_id()' method and I can adjust the code later to be faster

File: gbmodel/model_sqlalchemy.py [hunk 1]
```diff
@@ -8,6 +8,22 @@
 sys.path.append(os.getcwd())
 
 
+class professors(db.Model):
+    __table__ = db.Model.metadata.tables['professors']
+
+    # Get a list of professors
+    # Input: team id, session id
+    # Output: list of professors id
+    def get_professors(self, id):
```
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:17]: So this block comment doesn't correct

File: gbmodel/model_sqlalchemy.py [hunk 2]
```diff
@@ -270,20 +283,17 @@ def split_dates(self, params):
...
                 params[i] = params[i]
             else:
                 params[i] = None
-        session = capstone_session.query.filter(capstone_session.id == session_id).first()
-        session.midterm_start = params['midterm_start']
-        session.midterm_end = params['midterm_end']
-        session.final_start = params['final_start']
-        session.final_end = params['final_end']
-        db.session.commit()
+        engine.execute("update capstone_session set midterm_start =:midterm_start, \
```
Comment (J8cob1) [inline gbmodel/model_sqlalchemy.py:294]: Weren't we trying to switch over to the ORM? Maybe we can fix this later (if we want this in now)?

**Review by pdx-cs-lbryan — CHANGES_REQUESTED** [review]
* After a student has submitted their review, they will see 'generate reports' on the page.

* If a user goes to /review/ directly (without logging in), an exception occurs:
get() missing 1 required positional argument: 'user_id'

**Pull request closed without merging**